tests/Doctrine/Tests/Common/Annotations/DocParserTest.php

```diff
@@ -225,6 +226,123 @@ public function testTypicalMethodDocBlock(): void
         self::assertInstanceOf(Marker::class, $marker);
     }
 
+    public function testAnnotationWithConstructorWithVariadicParam(): void
+    {
+        $parser = $this->createTestParser();
+
+        // Without variadic arguments
+        $docblock = <<<'DOCBLOCK'
+/**
+ * @SomeAnnotationWithConstructorWithVariadicParam("Some data")
+ */
+DOCBLOCK;
+
+        $result = $parser->parse($docblock);
+        self::assertCount(1, $result);
+        $annot = $result[0];
+
+        self::assertInstanceOf(SomeAnnotationWithConstructorWithVariadicParam::class, $annot);
+
+        self::assertSame('Some data', $annot->name);
+        self::assertSame([], $annot->data);
+
+        // With named arguments
+        $docblock = <<<'DOCBLOCK'
+/**
+ * @SomeAnnotationWithConstructorWithVariadicParam(name = "Some data", foo = "Foo", bar = "Bar")
```

Comment:
This scenario must fail because the annotation does not have a parameter `$foo`.